lingua_franca/lang/common_data_az.py

```diff
@@ -0,0 +1,297 @@
+])
+
+
+_SHORT_SCALE_AZ = OrderedDict([
+    (100, 'yüz'),
+    (1000, 'min'),
+    (1000000, 'milyon'),
+    (1e9, "milyard"),
+    (1e12, 'trilyon'),
+    (1e15, "kvadrilyon"),
+    (1e18, "kvintilyon"),
+    (1e21, "sekstilyon"),
+    (1e24, "septilyon"),
+    (1e27, "oktilyon"),
+    (1e30, "nonilyon"),
+    (1e33, "dekilyon")
+])
+
+_ORDINAL_BASE_AZ = {
+    1: 'birinci',
+    2: 'ikinci',
+    3: 'üçüncü',
+    4: 'dördüncü',
+    5: 'beşinci',
+    6: 'altıncı',
+    7: 'yeddinci',
+    8: 'səkkizinci',
+    9: 'doqquzuncu',
+    10: 'onuncu',
+    11: 'on birinci',
+    12: 'on ikinci',
+    13: 'on üçüncü',
+    14: 'on dördüncü',
+    15: 'on beşinci',
+    16: 'on altıncı',
+    17: 'on yeddinci',
+    18: 'on səkkizinci',
+    19: 'on doqquzuncu',
+    20: 'iyirminci',
+    30: 'otuzuncu',
+    40: "qırxıncı",
+    50: "əllinci",
+    60: "altmışıncı",
+    70: "yetmışinci",
+    80: "səksəninci",
+    90: "doxsanınçı",
+    1e2: "yüzüncü",
+    1e3: "mininci"
+}
+
+_SHORT_ORDINAL_AZ = {
+    1e6: "milyonuncu",
+    1e9: "milyardıncı",
+    1e12: "trilyonuncu",
+    1e15: "kvadrilyonuncu",
+    1e18: "kvintilyonuncu",
+    1e21: "sekstilyonuncu",
+    1e24: "septilyonuncu",
+    1e27: "oktilyonuncu",
+    1e30: "nonilyonuncu",
+    1e33: "dekilyonuncu"
+    # TODO > 1e-33
+}
+_SHORT_ORDINAL_AZ.update(_ORDINAL_BASE_AZ)
+
+
+_LONG_ORDINAL_AZ = {
+    1e6: "milyonuncu",
+    1e12: "milyardıncı",
+    1e18: "trilyonuncu",
+    1e24: "kvadrilyonuncu",
+    1e30: "kvintilyonuncu",
+    1e36: "sekstilyonuncu",
+    1e42: "septilyonuncu",
+    1e48: "oktilyonuncu",
+    1e54: "nonilyonuncu",
+    1e60: "dekilyonuncu"
+    # TODO > 1e60
+}
+_LONG_ORDINAL_AZ.update(_ORDINAL_BASE_AZ)
+
+
+# negate next number (-2 = 0 - 2)
+_NEGATIVES_AZ = {"mənfi", "minus"}
+
+# sum the next number (twenty two = 20 + 2)
+_SUMS_AZ = {'on', '10', 'iyirmi', '20', 'otuz', '30', 'qırx', '40', 'əlli', '50',
+            'altmış', '60', 'yetmiş', '70', 'səksən', '80', 'doxsan', '90'}
+
+_HARD_VOWELS = ['a', 'ı', 'o', 'u']
+_SOFT_VOWELS = ['e', 'ə', 'i', 'ö', 'ü']
+_VOWELS = _HARD_VOWELS + _SOFT_VOWELS
+
+def _get_last_vowel(word):
+    is_last = True
+    for char in word[::-1]:
+        if char in _VOWELS:
+            return char, is_last
+        is_last = False
+
+    return "", is_last
+
+def _last_vowel_type(word):
+    return _get_last_vowel(word)[0] in _HARD_VOWELS
+
+def _get_ordinal_ak(word):
+    last_vowel, is_last = _get_last_vowel(word)
+    if not last_vowel:
+        return ""
+    
+    if last_vowel in ["a", "ı"]:
+        if is_last:
+            return "ncı"
+        return "ıncı"
+
+    if last_vowel == ["e", "ə", "i"]:
+        if is_last:
+            return "nci"
+        return "inci"
+
+    if last_vowel in ["o", "u"]:
+        if is_last:
+            return "ncu"
+        return "uncu"
+
+    if last_vowel == ["ö", "ü"]:
+        if is_last:
+            return "ncü"
+        return "üncü"
+
+def _get_full_time_ak(hour):
```

Comment:
what does "ak" refer to in these function names? I've seen it throughout this PR

Reply:
1) In Azerbaijani language instead of prepositions we have "ək"(I didn't want to use special characters in function name so i used "ak"). and the "ək" is added to the end of word. 
2) In Azerbaijani  we have two groups of vowels, and the "ək" used in word differs according to words last vowel and the ending of the word.

I'll explain with an example.
In AZ "five" is "beş" and "six" is "altı".
when saying time, in english both is same: at five, at six
but in Az "at five" is "beşdə"(the "ək" is "də") but "at six" is "altıda"(the "ək" is "da").

I use these functions to generate the correct ək for the words.

Reply:
Cool, thanks for the explanation - I'm always interested in seeing how different languages do things. Hopefully helps us to identify assumptions we make about the structure of all languages.